File: server/database/src/projections/spot/valut.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::types::Uuid;

use crate::types::Volume;

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Valut {
    pub id: Uuid,
    pub created_at: DateTime<Utc>,
    pub last_modification_at: DateTime<Utc>,
    pub user_id: Uuid,
    pub asset_id: Uuid,
    pub balance: Volume,
}

File: web/app/src/types/valut.ts
import { z } from "zod";
import { Uuid } from "./primitives/uuid";
import { Volume } from "./primitives/volume";
import { Datetime } from "./primitives/datetime";

export const Valut = z.object({
  id: Uuid,
  created_at: Datetime,
  last_modification_at: Datetime,
  user_id: Uuid,
  asset_id: Uuid,
  balance: Volume,
});
export type Valut = z.infer<typeof Valut>;
